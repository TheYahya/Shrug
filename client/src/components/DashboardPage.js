import React from 'react';
import NewUrl from './NewUrl';
import RecentUrls from './RecentUrls';
import { connect } from 'react-redux';
import QRCodeModal from './QRCodeModal';

const DashboardPage = (props) => {
  if (props.auth.isAuthenticated) {
    return (
        <div>
          <NewUrl />
          <RecentUrls /> 
          <QRCodeModal />
        </div>
    )
  }
  return (
    <div>
      <NewUrl />
    </div>
  )
}

const mapSteteToProps = (state) => {
  return {
    auth: state.auth
  };
};

export default connect(mapSteteToProps)(DashboardPage);