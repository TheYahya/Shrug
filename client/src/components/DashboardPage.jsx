import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import QRCodeModal from './QRCodeModal';
import NewUrl from './NewUrl';
import RecentUrls from './RecentUrls';

const DashboardPage = ({ auth }) => {
  if (auth.isAuthenticated) {
    return (
      <>
        <NewUrl />
        <RecentUrls />
        <QRCodeModal />
      </>
    );
  }
  return <NewUrl />;
};

const mapSteteToProps = (state) => ({
  auth: state.auth,
});

DashboardPage.propTypes = {
  auth: PropTypes.instanceOf(Object).isRequired,
};

export default connect(mapSteteToProps)(DashboardPage);
