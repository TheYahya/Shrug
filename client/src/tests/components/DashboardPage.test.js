import React from 'react';
import { shallow } from 'enzyme';
import DashboardPage from '../../components/DashboardPage';

test('should render DashboardPage correctly', () => {
  const wrapper = shallow(<DashboardPage />);
  expect(wrapper).toMatchSnapshot();
});
