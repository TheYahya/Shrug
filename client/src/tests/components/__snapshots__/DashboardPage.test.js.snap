// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DashboardPage correctly 1`] = `
<div>
  Dashboard page content
</div>
`;
